solver/src/solver.rs

```diff
@@ -18,15 +22,27 @@ arg_enum! {
     pub enum SolverType {
         Naive,
         UniswapBaseline,
+        MIP,
     }
 }
 
-pub fn create(solvers: Vec<SolverType>, base_tokens: HashSet<H160>) -> Vec<Box<dyn Solver>> {
+pub fn create(
+    solvers: Vec<SolverType>,
+    base_tokens: HashSet<H160>,
+    native_token: H160,
+    mip_solver_url: Url,
+) -> Vec<Box<dyn Solver>> {
     solvers
         .into_iter()
         .map(|solver_type| match solver_type {
             SolverType::Naive => Box::new(NaiveSolver {}) as Box<dyn Solver>,
             SolverType::UniswapBaseline => Box::new(UniswapSolver::new(base_tokens.clone())),
+            SolverType::MIP => Box::new(HttpSolver::new(
+                mip_solver_url.clone(),
+                None,
+                SolverConfig::new(100, 100),
```

Comment:
Is this time-limit in seconds? I think we should not give the solver more than 30s at the moment. Otherwise this can massively delay the other settlements as well given they wait on all to complete.